src/main/scala/com/fulcrumgenomics/util/MetricBuilder.scala

```diff
@@ -0,0 +1,142 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2022 Fulcrum Genomics
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ */
+
+package com.fulcrumgenomics.util
+
+import com.fulcrumgenomics.cmdline.FgBioMain.FailureException
+import com.fulcrumgenomics.commons.CommonsDef.{forloop, unreachable}
+import com.fulcrumgenomics.commons.reflect.{ReflectionUtil, ReflectiveBuilder}
+import com.fulcrumgenomics.commons.util.LazyLogging
+
+import java.io.{PrintWriter, StringWriter}
+import scala.reflect.runtime.{universe => ru}
+import scala.util.{Failure, Success}
+
+/** Class for building metrics of type [[T]].
+  *
+  * This is not thread-safe.
+  *
+  * @param source optionally, the source of reading (e.g. file)
+  * @tparam T the metric type
+  */
+class MetricBuilder[T <: Metric](source: Option[String] = None)(implicit tt: ru.TypeTag[T]) extends LazyLogging {
+  // The main reason why a builder is necessary is to cache some expensive reflective calls.
+  private val clazz: Class[T]   = ReflectionUtil.typeTagToClass[T]
+  private val reflectiveBuilder = new ReflectiveBuilder(clazz)
+  private val names             = Metric.names[T]
+
+  /** Builds a metric from a delimited line
+    *
+    * @param line       the line with delimited values
+    * @param delim      the delimiter of the values
+    * @param lineNumber optionally, the line number when building a metric from a line in a file
+    * @return
+    */
+  def fromLine(line: String, delim: String = Metric.DelimiterAsString, lineNumber: Option[Int] = None): T = {
+    fromValues(values = line.split(delim), lineNumber = lineNumber)
```

Comment:
Do you want to use [`StringUtil.split()`](https://github.com/fulcrumgenomics/commons/blob/e9534e08e61ac4da3d5ea8dd6f21a39b105005f2/src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala#L166) which could re-use an array and maybe be faster?